mattermosthooks.py

```diff
@@ -11,65 +13,117 @@
     'HgMattermostHooksConfig',
     field_names=[
         'webhook_url',
-        'repo_name',
+        'branches',
         'commit_url',
         'username',
         'icon_emoji',
         'icon_url',
     ])
 
 
-def get_config(ui):
-    webhook_url = ui.config(config_group, 'webhook_url')
-    repo_name = ui.config(config_group, 'repo_name', default=None)
-    commit_url = ui.config(config_group, 'commit_url', default=None)
-    username = ui.config(config_group, 'username', default="mercurial")
-    icon_emoji = ui.config(config_group, 'icon_emoji', default=None)
-    icon_url = ui.config(config_group, 'icon_url', default=None)
+def get_teams(ui):
+    yield 'default'
+    for field, value in ui.configitems(config_group):
+        if '.webhook_url' in field:
+            assert len(field) > len('.webhook_url'), 'Mattermost team configuration error'
+            yield field.split('.')[0]
 
-    return Config(webhook_url, repo_name, commit_url, username, icon_emoji,
-                  icon_url)
+
+def get_config(ui, team):
+    get_field = lambda field: '{team_with_separator}{field}'.format(
+        team_with_separator='' if team == 'default' else '{}.'.format(team),
+        field=field
+    )
+    return Config(
+        webhook_url = ui.config(
+            config_group, get_field('webhook_url')),
+
+        branches = ui.configlist(
+            config_group, get_field('branches')),
+
+        commit_url = ui.config(
+            config_group, 'commit_url', default=None),
+
+        username = ui.config(
+            config_group, 'username', default=ui.username()),
+
+        icon_emoji = ui.config(
+            config_group, 'icon_emoji', default=None),
+
+        icon_url = ui.config(
+            config_group, 'icon_url', default=None)
+    )
 
 
 def pushhook(node, hooktype, url, repo, source, ui, **kwargs):
-    username = ui.username()
-    config = get_config(ui)
+    '''Every team has its own configuration'''
+    for team in get_teams(ui):
+        publish(node, repo, ui, team)
+
 
-    changesets = get_changesets(repo, node)
+def publish(node, repo, ui, team):
+    config = get_config(ui, team)
+    changesets = get_changesets(repo, node, config.branches)
     count = len(changesets)
-    messages = render_changesets(ui, repo, changesets, config)
+    if not count:
+        return
+    # filter some characters that cause http 400 response
+    messages = re.sub(
+        '[;]', '',
```

Comment:
Our MatterMost hook breaks when markdown text like ```some info;``` appears in changeset description.